src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -132,24 +136,31 @@ grpc_channel_credentials* grpc_ssl_credentials_create(
 
   return grpc_core::New<grpc_ssl_credentials>(
       pem_root_certs, pem_key_cert_pair,
-      reinterpret_cast<const grpc_ssl_verify_peer_options*>(verify_options));
+      reinterpret_cast<const grpc_ssl_verify_peer_options*>(verify_options),
```

Comment:
Since we add a new field in grpc_ssl_verify_peer_options, I am worry that reinterpret_cast may break.
Previously grpc_ssl_server_verification_option is the same as server_verification_option, but a different name.

I think it is safer to have two constructors of grpc_ssl_credentials. 
One takes verify_peer_options, which corresponds to legacy grpc_ssl_credentials_create()
The other takes grpc_ssl_verify_peer_options and grpc_ssl_server_verification_option.
Then we need to set config_.verify_options accordingly.

Reply:
Added.